lib/request.js

```diff
@@ -319,7 +319,14 @@ module.exports = {
     var type = this.get('Content-Type');
     if (!type) return '';
 
-    return contentType.parse(type).parameters.charset || '';
+    try {
+      type = contentType.parse(type);
+    } catch (e) {
+      if (e instanceof TypeError) return '';
+      throw e
```

Comment:
`contentType.parse` will only throw `TypeError`, so `if (e instanceof TypeError)` is useless here, and line 326 will never be executed.

Even if `contentType.parse` throw some other unexpected errors, we also should return `""` here.